### PR TITLE
Send email when trainee assigned / removed

### PR DESCRIPTION
**Pull request by abdullahallmehedi**

### CODE REVIEW

**Comment by zenith19** [comment]
Ok

File: config/environments/development.rb [hunk 1]
```diff
@@ -38,4 +38,17 @@
...
   # Raises error for missing translations
   # config.action_view.raise_on_missing_translations = true
-end
+
+  config.action_mailer.smtp_settings = {
+        address:               "smtp.gmail.com",
+        port:                  587,
+        domain:                "gmail.com",
+        user_name:             "noreply.esv",
+        password:              "abrakada",
```
Comment (MuhammadTamzid) [inline config/environments/development.rb]: user_name and password should be stored in environment variable

File: app/mailers/user_mailer.rb [hunk 2]
```diff
@@ -10,4 +10,20 @@ def removed_supervisors user_full_name, user_email, course_id
     @user_full_name = user_full_name
     mail to: user_email, subject: default_i18n_subject(course_name: @course.name)
   end
+<<<<<<< HEAD
```
Comment (abdullahallmehedi) [inline app/mailers/user_mailer.rb]: remove

File: app/views/user_mailer/assigned_trainees.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  </head>
+  <body>
+    <h2>Hi!, <%= @user_full_name %></h2>
+    <h1>Welcome to <%= I18n.t :system_name %></h1>    
+    <p>
+      You have been assigned to a course.
+    </p>
+    <p>Course: <%= link_to @course.name, course_path(@course) %></p>
+  </body>
+</html>
```
Comment (abdullahallmehedi) [inline app/views/user_mailer/assigned_trainees.html.erb]: double quote

File: app/views/user_mailer/removed_trainees.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  </head>
+  <body>
+    <h2>Hi!, <%= @user_full_name %></h2>
+    <h1>Welcome to <%= I18n.t :system_name %></h1>    
+    <p>
+      You have been removed from a course.
+    </p>
+    <p>Course: <%= link_to @course.name, course_path(@course) %></p>
+  </body>
+</html>
```
Comment (abdullahallmehedi) [inline app/views/user_mailer/removed_trainees.html.erb]: double quote

File: app/models/user.rb [hunk 5]
```diff
@@ -19,6 +19,16 @@ class User < ActiveRecord::Base
 
   scope :supervisors, -> {where supervisor: true}
   scope :trainees, -> {where supervisor: false}
+  scope :course_unfinished,  -> {
```
Comment (dieunb) [inline app/models/user.rb]: Every time, user have only active course. So, i think you needn't define this. To find active course
```
user.courses.active.first
```

File: app/models/user.rb [hunk 6]
```diff
@@ -19,6 +19,16 @@ class User < ActiveRecord::Base
...
+    joins(:courses).where.not courses: {status: Course::STATUS[:finished]} 
+  }  
+  scope :assigned_to, ->(course) {
```
Comment (dieunb) [inline app/models/user.rb]: You can find course with id by:
```
user.courses.find course.id
```

File: app/models/user.rb [hunk 7]
```diff
@@ -19,6 +19,16 @@ class User < ActiveRecord::Base
...
+    joins(:courses).where "courses.id = ?", course.id
+  }
+  scope :assignable_trainees, ->(course) {
```
Comment (dieunb) [inline app/models/user.rb]: don't understand this scope!

File: app/models/user.rb (hunk 6 above)
Comment (abdullahallmehedi) [inline app/models/user.rb]: I mean to find all the users by course

File: app/models/user.rb (hunk 5 above)
Comment (abdullahallmehedi) [inline app/models/user.rb]: I am finding users (not courses) who have active courses.

Comment (dieunb) [inline app/models/user.rb]: ok, but bad naming, do you think about `free` name?

File: app/models/user.rb (hunk 6 above)
Comment (dieunb) [inline app/models/user.rb]: Do you think about? `course.users`

**Pull request merged**